types/codemirror/addon/hint/show-hint.d.ts

```diff
@@ -54,8 +54,13 @@ declare module "codemirror" {
     }
 
     interface ShowHintOptions {
-        completeSingle: boolean;
+        completeSingle?: boolean;
         hint: HintFunction | AsyncHintFunction;
+        alignWithWord?: boolean;
+        closeOnUnfocus?: boolean;
+        container?: Element | null;
+        customKeys?: {[x: string]: Function};
```

Comment:
https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes

The document says that `Using the types Function and Object is almost never a good idea.`.
Can't we specify the types of these functions?
I think we can specify the types as `() => void`, because the [`baseMap`](https://github.com/codemirror/CodeMirror/blob/5d740b99a4cca996b2cdc9fee3b1e1b8ac999617/addon/hint/show-hint.js#L159-L169) var has `() => void` functions.